packages/ember-metal/lib/property_get.js

```diff
@@ -52,11 +52,22 @@ export function get(obj, keyName) {
   assert(`The key provided to get must be a string, you passed ${keyName}`, typeof keyName === 'string');
   assert(`'this' in paths is not supported`, !pathHasThis(keyName));
 
-  // Helpers that operate with 'this' within an #each
-  if (keyName === '') {
+  let type = typeof obj;
+  let isObject = obj !== null && type === 'object';
+  let isFunc = type === 'function';
+  let isStringLength = type === 'string' && keyName === 'length';
+
+  if (isObject || isFunc || isStringLength) {
+    return _get(obj, keyName);
+  } else if (keyName === '') {
```

Comment:
i would prefer to inline the expression, so we don't need to evaluate all of them. Ultimately the JIT/uglifer may be smart enough to inline this...